pkg/specgen/generate/namespaces_linux.go

```diff
@@ -98,8 +102,13 @@ func specConfigureNamespaces(s *specgen.SpecGenerator, g *generate.Generator, rt
 		// the user or if we are creating a new UTS namespace.
 		// TODO: Should we be doing this for pod or container shared
 		// namespaces?
+		// we do not want to add domainname to the hostname, we need to keep track of it for later
 		g.SetHostname(hostname)
 	}
+
+	if s.Domainname != "" || s.UtsNS.NSMode != specgen.Host {
```

Comment:
Should this be an &&?

Otherwise you could be setting Domain name to ""?

Reply:
right, I will change that